### PR TITLE
Changed outdated type-function to is_<type> 

### PR DESCRIPTION
**Pull request by ejabach**
Also set stdlib version to 4.2 and adjusted spec test to match fail messages.

### CODE REVIEW

**Comment by Phil-Friderici** [comment]
looks good to me, thanks @ejabach

File: spec/classes/agent_spec.rb [hunk 1]
```diff
@@ -590,7 +590,7 @@
       it 'should fail' do
         expect {
           should contain_class('puppet::agent')
-        }.to raise_error(Puppet::Error,/^undefined method `to_i'/)
+        }.to raise_error(Puppet::Error,/puppet::agent::puppet_masterport is set to <invalidtype>. It should be an integer./)
```
Comment (ghoneycutt) [inline spec/classes/agent_spec.rb]: period `.` acts like a `*` in a regex and matches everything. They should be escaped such as `/This is how to handle a period\. It has a slash in front of it\./`

**Comment by Phil-Friderici** [comment]
@ghoneycutt I am taking over this in https://github.com/ghoneycutt/puppet-module-puppet/pull/111

File: spec/classes/agent_spec.rb (hunk 1 above)
Comment (Phil-Friderici) [inline spec/classes/agent_spec.rb]: period matches only one character, but I see your point (hahahaha) here

**Pull request closed without merging**